src/extension.ts

```diff
@@ -34,31 +34,51 @@ export function activate(context: vscode.ExtensionContext) {
 
     context.subscriptions.push(disposable1);
 
-    const disposable2 = vscode.commands.registerCommand('ucl-rsdg.validateAtf', async () => {
-        const editor = vscode.window.activeTextEditor;
-        const fileName = basename(editor.document.uri.fsPath);
-        const fileContent = editor.document.getText();
-        let fileProject: string;
-        try {
-            fileProject = getProjectCode(fileContent);
-        } catch (err) {
-            vscode.window.showErrorMessage(`Could not validate: ${err}`);
-            return;
-        }
-        // The validate function is currently not mapped to the appropriate logging functions
-        const result = await validate(fileName,fileProject,fileContent);
-        handleResult(result, editor);
+    const disposable2 = vscode.commands.registerCommand('ucl-rsdg.validateAtf', () => {
```

Comment:
why the odd variable name? and shouldn't `disposable1` be... disposed of now?

Reply:
Those were the names before. It's because of the pattern used: you create the subscription, then add it to the list of subscriptions (to be disposed when the extension is deactivated, I think). The code for the new command uses a simpler pattern that avoids the intermediate variable. We can change the older ones to that too.